torch/_functorch/_aot_autograd/subclass_utils.py

```diff
@@ -24,10 +24,10 @@ def requires_subclass_dispatch(args, fw_metadata: ViewAndMutationMeta) -> bool:
         for x in args_flattened
         if isinstance(x, Tensor)
     )
+    from torch._functorch._aot_autograd.schemas import SubclassCreationMeta
+
     any_subclass_outputs = any(
-        is_traceable_wrapper_subclass(x)
-        for x in fw_metadata.traced_tangents
-        if isinstance(x, Tensor)
+        type(x) is SubclassCreationMeta for x in fw_metadata.subclass_fw_graph_out_meta
```

Comment:
Noob q; 
From reading the issue you linked,  I take it that the problem wasn't really the  "check" for subclass outputs but really that we weren't checking enough things?

For the check on inputs is that exhaustive enough?

Reply:
> I take it that the problem wasn't really the "check" for subclass outputs but really that we weren't checking enough things?

Yep this is right

> For the check on inputs is that exhaustive enough?

yeah - the reason the check wasn't exhaustive in the first place is because "graph outputs" and "tangents" can be different (there can be extra outputs to the forward graph, that don't get tangents). But "graph inputs" doesn't have any extra complexity to it - we just loop through them all right now